javascript/extractor/src/com/semmle/js/extractor/AutoBuild.java

```diff
@@ -444,33 +449,119 @@ protected boolean hasSeenCode() {
 
   /** Perform extraction. */
   public int run() throws IOException {
-    startThreadPool();
-    try {
-      CompletableFuture<?> sourceFuture = extractSource();
-      sourceFuture.join(); // wait for source extraction to complete
-      if (hasSeenCode()) { // don't bother with the externs if no code was seen
-        extractExterns();
+    initDiagnosticsWriter();
+
+    try (DiagnosticWriter writerToClose = diagnostics) {
+      startThreadPool();
+      try {
+        CompletableFuture<?> sourceFuture = extractSource();
+        sourceFuture.join(); // wait for source extraction to complete
+        if (hasSeenCode()) { // don't bother with the externs if no code was seen
+          extractExterns();
+        }
+        extractXml();
+      } catch (OutOfMemoryError oom) {
+        return 137; // the CodeQL CLI will interpret this as an out-of-memory error
```

Comment:
Let's at least try to print a constant string to`System.err` like you do in the `doExtract` method below.